src/hotspot/share/code/codeCache.cpp

```diff
@@ -368,7 +368,7 @@ bool CodeCache::heap_available(int code_blob_type) {
   if (!SegmentedCodeCache) {
     // No segmentation: use a single code heap
     return (code_blob_type == CodeBlobType::All);
-  } else if (Arguments::is_interpreter_only()) {
+  } else if (CompilerConfig::is_interpreter_only()) {
```

Comment:
Doesn't this introduce a very subtle change in behaviour in relation to the handling of `MethodNonProfiled`?
@vnkozlov please confirm this is what you intended - I don't know the implications.

Reply:
According to my conversation with Rickard, TierStopAtLevel=0 is the same as -Xint.  If TieredStopAtLevel=0, the compiler won't allocate nmethods in the MethodProfiled and MethodNonProfiled areas anyway, so this won't allocate them now.

Reply:
I think this code was missed during Igor's V. changes which introduced `CompilerConfig::is_interpreter_only()`.
Coleen is correct, with `TieredStopAtLevel=0` all JIT compilers are disabled. `MethodNonProfiled` is codeheap for tier1 (C1) and tier4 (C2) compiled nmethods. It does not make sense to have such codeheap when compilation disabled.

Reply:
Okay, so does that mean this block:
```
} else {
    // No TieredCompilation: we only need the non-nmethod and non-profiled code heap
    return (code_blob_type == CodeBlobType::NonNMethod) ||
           (code_blob_type == CodeBlobType::MethodNonProfiled);
}
```
is actually unreachable?

Reply:
It is reachable for `-XX:-TieredCompilation` when only C2 is used. Or when only tier1 C1 is used with `-XX:TieredStopAtLevel=1`.

Reply:
Ah I see. Thanks for clarifying @vnkozlov !

Reply:
-XX:TieredStopAtLevel=1 is the old -client switch. Thanks Vladimir for confirming.